### PR TITLE
Use golang 1.16 embed instead of our own

### PR DESCRIPTION
**Pull request by skitt**
Embedded files are always relative to the package embedding them, and
can't be symlinks, so we need to move config and deploy.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

File: pkg/yamls/yamls.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package embeddedyamls
+
+import _ "embed"
+
+//go:embed deploy/crds/submariner.io_brokers.yaml
```
Comment (Jaanki) [inline pkg/yamls/yamls.go]: Embed all files into a file system.

Eg:

import "embed"
// content is our static web server content.
//go:embed image template html/index.html
var content embed.FS

**Review by Jaanki — CHANGES_REQUESTED** [review]

**Comment by mangelajo** [comment]
Cool

File: pkg/yamls/yamls.go (hunk 1 above)
Comment (Jaanki) [reply pkg/yamls/yamls.go]: The calling functions expect a string. So embed them as a string as you are currently doing.

**Review by Jaanki — CHANGES_REQUESTED** [review]
`submariner-operator/pkg/subctl/operator/common/embeddedyamls/generators/yamls2go.go` needs to be deleted

File: Makefile [hunk 2]
```diff
@@ -129,27 +135,33 @@ bin/subctl-%: generate-embeddedyamls $(shell find pkg/subctl/ -name "*.go") vend
...
+# There's a chicken-and-egg situation here: we need the embedded resources
+# to build anything, including the builds required by the controller; so
+# we touch the generated files before calling the generator, and delete
+# empty files afterwards
+
+pkg/yamls/deploy/crds/submariner.io_servicediscoveries.yaml: $(CONTROLLER_GEN) ./apis/submariner/v1alpha1/servicediscovery_types.go vendor/modules.txt
+	for file in $(GENERATED_YAMLS); do mkdir -p $${file%/*}; done
+	touch $(GENERATED_YAMLS)
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) paths=./... output:crd:artifacts:config=pkg/yamls/deploy/crds
+	find $(GENERATED_YAMLS) -empty -delete
```
Comment (skitt) [inline Makefile]: This is ugly and duplicated below, I’ll fix it.

File: pkg/yamls/yamls.go (hunk 1 above)
Comment (skitt) [reply pkg/yamls/yamls.go]: Right, and this would also avoid the chicken-and-egg problem...

**Comment by Jaanki** [comment]
bump

**Comment by Jaanki** [comment]
reopen

**Pull request closed without merging**